src/core/dev/Encoder.cpp

```diff
@@ -114,17 +114,23 @@ void Encoder::setRangeAndPosition(uint32_t newRange, uint32_t newPosition) {
 }
 
 int8_t Encoder::readPinValues() {
-    bool aPos = (bool) a.readPin();
-    bool bPos = (bool) b.readPin();
-    // calculating and returning the position
-    if (aPos == 0 && bPos == 0) {
+    uint8_t aPos = (bool) a.readPin();
```

Comment:
Why did you do this?

Reply:
The switch statement is more efficient, personally I think more readable, and having the default case removes the compiler warning that the method may not return a value.

Reply:
Can you add a comment here explaining why you're keeping it as a `uint8_t`? Your explanation makes sense, but it's not immediately obvious. Also, you may as well not cast to `bool` here because you're immediately converting to `uint8_t`

Reply:
Cleaned it up and made it a bit more obvious what is happening.